util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/dialect/ElasticsearchTestDialect.java

```diff
@@ -6,86 +6,69 @@
 
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isBetween;
+import static org.hibernate.search.util.impl.integrationtest.backend.elasticsearch.dialect.ElasticsearchVersionUtils.isLessThan;
+
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 import java.util.Optional;
 
 import org.hibernate.search.backend.elasticsearch.ElasticsearchVersion;
-import org.hibernate.search.backend.elasticsearch.client.spi.ElasticsearchRequest;
+import org.hibernate.search.backend.elasticsearch.client.impl.Paths;
 import org.hibernate.search.backend.elasticsearch.util.spi.URLEncodedString;
 
-import com.google.gson.JsonObject;
+public class ElasticsearchTestDialect {
 
-public interface ElasticsearchTestDialect {
+	private static final ElasticsearchVersion ACTUAL_VERSION = ElasticsearchVersion.of(
+			System.getProperty( "org.hibernate.search.integrationtest.backend.elasticsearch.version" )
+	);
 
-	static ElasticsearchTestDialect get() {
-		String dialectClassName = System.getProperty( "org.hibernate.search.integrationtest.backend.elasticsearch.testdialect" );
-		try {
-			@SuppressWarnings("unchecked")
-			Class<? extends ElasticsearchTestDialect> dialectClass =
-					(Class<? extends ElasticsearchTestDialect>) Class.forName( dialectClassName );
-			return dialectClass.getConstructor().newInstance();
-		}
-		catch (Exception | LinkageError e) {
-			throw new IllegalStateException(
-					"Unexpected error while initializing the ElasticsearchTestDialect with name '" + dialectClassName + "'."
-							+ " Did you properly set the appropriate elasticsearch-x.x/opensearch-x.x profile?",
-					e
-			);
-		}
-	}
+	private static final ElasticsearchTestDialect INSTANCE = new ElasticsearchTestDialect();
 
-	static ElasticsearchVersion getActualVersion() {
-		return ElasticsearchVersion.of( System.getProperty( "org.hibernate.search.integrationtest.backend.elasticsearch.version" ) );
+	public static ElasticsearchTestDialect get() {
+		return INSTANCE;
 	}
 
-	boolean isEmptyMappingPossible();
-
-	URLEncodedString getTypeKeywordForNonMappingApi();
+	public static ElasticsearchVersion getActualVersion() {
+		return ACTUAL_VERSION;
+	}
 
-	Optional<URLEncodedString> getTypeNameForMappingAndBulkApi();
+	public boolean isEmptyMappingPossible() {
+		return isLessThan( ACTUAL_VERSION, "elastic:6.9.0" );
+	}
 
-	Boolean getIncludeTypeNameParameterForMappingApi();
+	@SuppressWarnings("deprecation")
+	public Optional<URLEncodedString> getTypeNameForMappingAndBulkApi() {
+		if ( isLessThan( ACTUAL_VERSION, "elastic:6.9.0" ) ) {
+			return Optional.of( Paths.DOC );
+		}
+		return Optional.empty();
+	}
 
-	ElasticsearchRequest createTemplatePutRequest(String templateName, String pattern, int priority, JsonObject settings);
+	public Boolean getIncludeTypeNameParameterForMappingApi() {
+		if ( isBetween( ACTUAL_VERSION, "elastic:6.4.0", "elastic:6.9.0" ) ) {
+			return Boolean.TRUE;
```

Comment:
This seems wrong? From what I can see, we were returning true for `Elasticsearch68TestDialect` only, which means for ES 6.8 and 6.9 only.

I think this should be:

```suggestion
		if ( isBetween( ACTUAL_VERSION, "elastic:6.8", "elastic:6.9" ) ) {
			return Boolean.TRUE;
```

Reply:
ooooooooooooooooook 🥲 so this was 
- `null` in 64 dialect 
- inherited `true` in 67 dialect 
- `true` in 68 dialect 
- `null` in 70-712 dialects 
but then in the pom ... 6.4 is actually a 6.6. Which ... I suppose, this converts the condition to `isBetween( ACTUAL_VERSION, "elastic:6.6", "elastic:6.9" )` 
running the tests for 6.4 6.7 and 6.8 profiles now 🥲

Reply:
>  but then in the pom ... 6.4 is actually a 6.6. Which ... I suppose, this converts the condition to isBetween( ACTUAL_VERSION, "elastic:6.6", "elastic:6.9" )

Don't you mean `isBetween( ACTUAL_VERSION, "elastic:6.7", "elastic:6.9" )`? Since 6.6 was being tested with the 6.4 profile, which uses `null`?